testgen/generators/plurals.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+import os
+import json
+import re
+import logging
+import subprocess
+import xml.etree.ElementTree as ET
+
+from generators.base import DataGenerator
+
+reblankline = re.compile("^\s*$")
+
+class PluralGenerator(DataGenerator):
+    def plurals_descriptor(self):
+        # Create test list of all these
+        self.test_id = 'plural_rules'
+        self.categories = ['zero', 'one', 'two', 'few', 'many', 'other']
+
+        source_url = ['https://github.com/unicode-org/cldr/blob/main/common/supplemental/plurals.xml',
```

Comment:
Actually, I think you want to replace the word `main` with the release label (ex: `release-41`) in order for the CLDR data to match the version of ICU that this version platform is based on, right?

So if we're talking about generating ICU 71 data, then CLDR version is 41, so we want `'https://github.com/unicode-org/cldr/blob/release-41/common/supplemental/plurals.xml'`, etc.

If we don't change the word `main` in those URLs, then I would predict that all of our generated data files would be identical across ICU versions, and the contents of those data files would depend on what CLDR looked like at the time that we ran the data generation step most recently.

Reply:
This is a good catch. This source_url field isn't used to fetch the data, but is just a descriptor. But it should be accurate. I'll try to update this.